engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -263,29 +268,59 @@ private void processAggregateEmailsByAggregationKey(EmailAggregationKey aggregat
         }
 
         if (subject != null && body != null) {
-            for (Map.Entry<User, Map<String, Object>> aggregation :
-                    emailAggregator.getAggregated(aggregationKey, emailSubscriptionType, startTime, endTime).entrySet()) {
+            Map<User, Map<String, Object>> aggregationsByUsers = emailAggregator.getAggregated(aggregationKey, emailSubscriptionType, startTime, endTime);
+
+            if (featureFlipper.isSendSingleEmailForMultipleRecipientsEnabled()) {
+                Map<Map<String, Object>, Set<User>> aggregationsEmailContext = aggregationsByUsers.keySet().stream()
+                    .collect(Collectors.groupingBy(aggregationsByUsers::get, Collectors.toSet()));
```

Comment:
I'm okay with that as a temporary solution but we'll need to make the whole aggregation process a lot simpler than it is now. My comment is not related to your PR, I already didn't like that process before 😅

Reply:
I fully agree, the full aggregation process should be simpler